Marlin/Marlin_main.cpp

```diff
@@ -284,11 +284,11 @@ bool axis_homed[3] = { false };
 
 static long gcode_N, gcode_LastN, Stopped_gcode_LastN = 0;
 
-static char* current_command, *current_command_args;
-static int cmd_queue_index_r = 0;
-static int cmd_queue_index_w = 0;
-static int commands_in_queue = 0;
 static char command_queue[BUFSIZE][MAX_CMD_SIZE];
+static char* current_command, *current_command_args;
+static uint8_t cmd_queue_index_r = 0,
+               cmd_queue_index_w = 0,
+               commands_in_queue = 0;
```

Comment:
Why not volatile?
`loop()` might load `commands_in_queue` into a register. When calling `process_next_command()` this register might saved to the stack. `process_next_command()` might try to store a move in the planner. If the planner buffer is full it calls `idle()`, calling `manage_inactivity()`, calling `get_available_commands()`. That may add a a new command into the buffer - `commands_in_queue` is increased in memory. All called functions may return - the registers restored and `loop()` may have a wrong value for `commands_in_queue` now, what is decreased and stored back to memory.

A static volatile uint8_t  at least can not hurt.